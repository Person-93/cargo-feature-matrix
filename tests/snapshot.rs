use assert_cmd::Command;
use insta::{assert_snapshot, glob, with_settings};
use lazy_static::lazy_static;
use std::{env, path::PathBuf};

#[test]
fn test() {
    glob!("samples/*/Cargo.toml", |path| {
        let path = path
            .parent()
            .expect("failed to unwrap parent of Cargo.toml");

        let mut cmd = Command::new(BIN.as_os_str());
        let assert = cmd
            .arg("feature-matrix")
            .arg("--color=never")
            .arg("--dry-run")
            .arg("clippy")
            .current_dir(path)
            .assert()
            .success();
        let output = assert.get_output();
        let output = std::str::from_utf8(&output.stdout)
            .expect("child process's output included invalid unicode");

        with_settings!({snapshot_suffix => path.file_name().unwrap().to_str().unwrap()},
        {
            assert_snapshot!(output);
        });
    });
}

lazy_static! {
    static ref BIN: PathBuf =
        assert_cmd::cargo::cargo_bin("cargo-feature-matrix");
}
