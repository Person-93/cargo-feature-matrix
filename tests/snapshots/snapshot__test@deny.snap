---
source: tests/snapshot.rs
expression: output

---
running: cmd=clippy package=sample features=[]......OK
running: cmd=clippy package=sample features=[feat-a,feat-b]......OK
running: cmd=clippy package=sample features=[feat-b]......OK
